src/target/efm32.c

```diff
@@ -538,12 +544,13 @@ bool efm32_probe(target_s *t)
 {
 	/* Check if the OUI in the EUI is silabs or energymicro.
 	 * Use this to identify the Device Identification (DI) version */
-	uint8_t di_version = 1;
-	uint64_t oui24 = ((efm32_v1_read_eui64(t) >> 40U) & 0xffffffU);
-	if (oui24 == EFM32_V1_DI_EUI_SILABS) {
+	uint8_t di_version;
+	uint64_t oui24_v1 = ((efm32_v1_read_eui64(t) >> 40U) & 0xffffffU);
+	uint64_t oui24_v2 = ((efm32_v2_read_eui48(t) >> 24U) & 0xffffffU);
```

Comment:
These can be made `const`, so let's do so. They also look to have masks that reduce them to 32 bits or less, so let's have them as `uint32_t`'s - this will improve codegen a bit. The outer parens are unnecessary and can be dropped.